### PR TITLE
AAP-27695 Add Ansible plug-in for RHDH Installation doc

### PR DESCRIPTION
**Pull request by ariordan-redhat**
[AAP-27695](https://issues.redhat.com/browse/AAP-27695) Add Ansible plug-in for RHDH Installation doc.

[Preview build](https://ariordan-redhat.github.io/aap-builds/aap-plugin-rhdh-install-aap-27695-2024-07-25.html)

Affects `titles/aap-plugin-rhdh-install`

### CODE REVIEW

File: downstream/modules/devtools/ref-rhdh-install-ocp-prereqs.adoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+:_mod-docs-content-type: REFERENCE
+
+[id="rhdh-install-ocp-prereqs_{context}"]
+= Prerequisites
+
+* {RHDH} installed on {OCP} using a Helm chart.
+Follow the steps in the link:{BaseURL}/red_hat_developer_hub/1.2/html-single/getting_started_with_red_hat_developer_hub/proc-install-rhdh-helm_rhdh-getting-started[Deploying Red Hat Developer Hub on OpenShift Container Platform] chapter of _Administration Guide for Red Hat Developer Hub_.
```
Comment (g-murray) [inline downstream/modules/devtools/ref-rhdh-install-ocp-prereqs.adoc:7]: The destination of this link doesn't match with either of the titles which can be confusing for the user. The correct title is used in your google doc ("Getting started with Red Hat Developer Hub")

File: downstream/modules/devtools/proc-rhdh-create-plugin-registry.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+$ oc new-app --image-stream=plugin-registry
+----
+
+.Verification
+
+To verify that the plugin-registry was deployed successfully, open the Developer perspective on the {RHDH} application in the OpenShift Web console.
+
+. Click the plug-in registry to view the log. 
++
+image::rhdh-plugin-registry.png[Developer perspective]
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-create-plugin-registry.adoc:30]: This image doesn't show in your preview PR, but that might expect behaviour for a pr-preview?

File: downstream/modules/devtools/proc-rhdh-add-plugin-config.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          http://plugin-registry:8080/ansible-plugin-backstage-rhaap-backend-x.y.z.tgz
+        pluginConfig:
+          dynamicPlugins:
+            backend:
+              ansible.plugin-backstage-rhaap-backend: null
+
+----
+. In the `package` sections, replace `x.y.z`  in the plug-in filenames with the correct version numbers for the Ansible plug-ins.
+. For each Ansible plug-in, update the integrity values using the corresponding `.integrity` file content.
+. Click btn[Upgrade].
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-add-plugin-config.adoc:51]: you're missing the colon. btn:[Upgrade]

File: downstream/modules/devtools/proc-rhdh-add-plugin-config.adoc [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+The developer hub pods restart and the plug-ins are installed.
+
+.Verification
+
+Verify that the plug-ins have been installed:
+
+. Open the Developer perspective for the {RHDH} application in the OpenShift Web console.
+. Check the `install-dynamic-plugin` container logs: the Ansible plug-ins are visible in {RHDH}.
++
+image::rhdh-check-plugin-config.png[container logs for install-dynamic-plugin]
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-add-plugin-config.adoc:62]: Image not displaying but that might just eb cause this is a preview pr?

File: downstream/modules/devtools/proc-rhdh-add-devtools-container.adoc [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+The image is pulled only if it does not already exist on the node.
+Update it to `imagePullPolicy: Always` if you always want to use the latest image.
+====
+. Click btn:[Upgrade].
+
+.Verification
+
+To verify that the container is running, check the container log:
+
+image::rhdh-check-devtools-container.png[View container log]
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-add-devtools-container.adoc:50]: same comment as other images

File: downstream/modules/devtools/proc-rhdh-configure-aap-details.adoc [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+[NOTE]
+====
+The Ansible plug-ins continue to function regardless of the {PlatformNameShort} subscription status.
+====
+
+.Procedure
+
+. Create a Personal Access Token (PAT) with “Read” scope in automation controller, following the
+link:{BaseURL}/red_hat_ansible_automation_platform/{PlatformVers}/html/automation_controller_user_guide/index#assembly-controller-applications[Applications] section of the _Automation controller user guide_.
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-configure-aap-details.adoc:16]: This link doesn't work for me but I can't see whats (if anythings) wrong with it?

File: downstream/modules/devtools/proc-rhdh-warning-unable-authenticate-aap.adoc [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+Verify that the authentication details for Ansible Automation Platform are correctly configured in the Ansible plug-ins.
+For help, please refer to the Ansible plug-ins installation guide.
+----
+
+[discrete]
+== Remediation steps
+
+. Verify that the automation controller Personal Access Token (PAT) configured in the Ansible plug-ins is correct.
+For more information, refer to the
+link:{BaseURL}/red_hat_ansible_automation_platform/{PlatformVers}/html/automation_controller_user_guide/index#assembly-controller-applications[Applications]
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-warning-unable-authenticate-aap.adoc:19]: Same as before link doesn't work in the preview but I can't see anythingw wrong?

File: downstream/modules/devtools/proc-rhdh-update-plugins-version-numbers.adoc [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+              ansible.plugin-backstage-rhaap-backend: null
+
+----
+. Click btn:[Upgrade].
++
+The developer hub pods restart and the plug-ins are installed.
+
+.Verification
+
+. In the OpenShift UI, click *Topology*.
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-update-plugins-version-numbers.adoc:57]: I expect this is a tab after our conversation so all good just using asterisks

File: downstream/modules/devtools/proc-rhdh-configure-optional-integrations.adoc [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+[id="rhdh-configure-optional-integrations_{context}"]
+= Configuring Ansible plug-ins optional integrations
+
+The Ansible plug-ins provide integrations with {PlatformNameShort} and other optional Red Hat products.
+
+Create a custom ConfigMap called `app-config-rhdh` as outlined in the
+link:{BaseURL}/red_hat_developer_hub/1.2/html-single/administration_guide_for_red_hat_developer_hub/assembly-install-rhdh-ocp#proc-add-custom-app-file-openshift-helm_assembly-install-rhdh-ocp[Adding a custom application configuration file to OpenShift Container Platform using the Helm chart] of the _Administration guide for Red Hat Developer Hub_.
+
+To edit your ConfigMap, log in to the OpenSHift UI and navigate to menu:Select Project ( developerHubProj )[ConfigMaps > {developer-hub}-app-config > EditConfigMaps > app-config-rhdh].
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-configure-optional-integrations.adoc:11]: Typo: The "h" in OpenShift is upper case

File: downstream/modules/devtools/proc-rhdh-configure-pah-url.adoc [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+====
+The {PrivateHubName} configuration is optional but recommended.
+The Ansible plug-ins will function without it.
+====
+
+.Prerequisites: 
+* A {PrivateHubName} instance.
++
+For more information on installing {PrivateHubName},  refer to the
+link:{BaseURL}/red_hat_ansible_automation_platform/{PlatformVers}#Installation%20and%20Upgrade[Installaion and Upgrade]
```
Comment (g-murray) [inline downstream/modules/devtools/proc-rhdh-configure-pah-url.adoc:21]: This link also leads me to a 404 error

**Review by g-murray — COMMENTED** [review]
Hey Aine, two or 3 small typos, and some of the links (and images) didn't work for me but I couldn't see anything wrong with them so I'm wondering if its a preview-pr limitation?

File: downstream/modules/devtools/proc-rhdh-add-plugin-config.adoc (hunk 4 above)
Comment (ariordan-redhat) [reply downstream/modules/devtools/proc-rhdh-add-plugin-config.adoc:62]: @g-murray Yes, I didn't add the images to the preview repo.
The images look fine locally.

**Comment by ariordan-redhat** [comment]
Need to update some links in main (to controller doc in particular) before this can be backported to 2.5

**Pull request merged**